src/kagglehub/models.py

```diff
@@ -36,18 +39,22 @@ def model_upload(
         license_name: (string) model license.
         version_notes: (string) Optional to write to model versions.
     """
-    # parse slug
-    h = parse_model_handle(handle)
+    try:
+        # parse slug
+        h = parse_model_handle(handle)
 
-    if h.is_versioned():
-        is_versioned_exception = "The model handle should not include the version"
-        raise ValueError(is_versioned_exception)
+        if h.is_versioned():
+            is_versioned_exception = "The model handle should not include the version"
+            raise ValueError(is_versioned_exception)
 
-    # Create the model if it doesn't already exist
-    create_model_if_missing(h.owner, h.model)
+        # Create the model if it doesn't already exist
+        create_model_if_missing(h.owner, h.model)
 
-    # Upload the model files to GCS
-    tokens = upload_files(local_model_dir, "model")
+        # Upload the model files to GCS
+        tokens = upload_files(local_model_dir, "model")
 
-    # Create a model instance if it doesn't exist, and create a new instance version if an instance exists
-    create_model_instance_or_version(h, tokens, license_name, version_notes)
+        # Create a model instance if it doesn't exist, and create a new instance version if an instance exists
+        create_model_instance_or_version(h, tokens, license_name, version_notes)
+    except KeyboardInterrupt:
+        delete_model(h.owner, h.model, h.framework, h.variation, h.version)
```

Comment:
This can have unintended consequences. I rather not delete unless explicitly invoked by the user. 

Example scenarios:
* Midtier is updated and the logic cascades to delete the entire model page.

Reply:
We should update the `create_model_instance_or_version` method to idempotent on the midtier side.

Reply:
I think i will remove the delete mode logic from this cl to make it easier to review and I can make a pr for it later